op/model.py

```diff
@@ -72,6 +71,38 @@ def __init__(self, name, backend, cache):
 
         self._is_our_app = self.name == self._backend.app_name
 
+        self._status = None
+
+    @property
+    def status(self):
+        if not self._is_our_app:
+            return UnknownStatus()
+
+        if not self._backend.is_leader():
+            raise RuntimeError('cannot get application status as a non-leader unit')
+
+        if self._status:
+            return self._status
+
+        s = self._backend.status_get(True)
+        self._status = Status.from_string(s['status'], s['message'])
+
+        return self._status
```

Comment:
Empty line above may be dropped as well. Sorry for the nitpick.. it just feels like we're having sparse line layouts too often without actually being helpful, so I'm trying to balance it out a bit.